ibm_code_engine_app: Plugin crashes when updating Code Engine Apps

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform IBM Provider Version

Terraform v1.6.6

Provider v1.74.0

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* ibm_code_engine_app

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Configuration of environment variables before error message:

```hcl
resource "ibm_code_engine_app" "code_engine_app_instance" {
  project_id      = ibm_code_engine_project.code_engine_project_instance.project_id
  name            = "my-app"
  image_reference = "icr.io/codeengine/helloworld"

  run_env_variables {
    type  = "literal"
    name  = "foo"
    value = "foo"
  }

  run_env_variables {
    type  = "literal"
    name  = "bar"
    value = "bar"
  }
}
```

Configuration of environment variables that leads to error message:

```hcl
resource "ibm_code_engine_app" "code_engine_app_instance" {
  project_id      = ibm_code_engine_project.code_engine_project_instance.project_id
  name            = "my-app"
  image_reference = "icr.io/codeengine/helloworld"

  run_env_variables {
    type  = "literal"
    name  = "foo"
    value = "foo"
  }

  run_env_variables {
    type  = "literal"
    name  = "foobar"
    value = "foobar"
  }

  run_env_variables {
    type  = "literal"
    name  = "bar"
    value = "bar"
  }
}
```

### Debug Output

See [Gist](https://gist.github.com/team-geodserv/8d3f870206d2e2f9a2bde2e87a599b75) to show debug output

### Panic Output

```hcl
panic: interface conversion: interface {} is []map[string]interface {}, not []interface {}
```

### Expected Behavior

Insert an enviroment variable into the list of existing ones (see example: insert at index 1, elements on index 0, 1 already available) and add this to the accessable variables in the Code Engine App.

### Actual Behavior

Plan can not be applied. Terraform crashed into the linked error and panic output when trying to insert an environment variable.

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform plan`
2. `terraform apply`

### Important Factoids

Internally we use modules to configure ressources. In the above shown examples you do not see such a module but a plain terraform config. The debug output in our gist shows the error results given by the execution via module.

Another very similar crash:

` 2025/02/20 05:50:01 Terraform apply | 2025-02-20T05:50:01.657Z [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: EOF"
 2025/02/20 05:50:01 Terraform apply | 2025-02-20T05:50:01.658Z [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: EOF"
 2025/02/20 05:50:01 Terraform apply | 2025-02-20T05:50:01.710Z [ERROR] vertex "module.code_engine_instance.ibm_code_engine_config_map.app_config_map[\"0\"]" error: Plugin did not respond
 2025/02/20 05:50:01 Terraform apply | 2025-02-20T05:50:01.741Z [ERROR] vertex "module.code_engine_instance.ibm_code_engine_app.app[\"0\"]" error: Plugin did not respond
 2025/02/20 05:50:01 Terraform apply | 
 2025/02/20 05:50:01 Terraform apply | Error: Plugin did not respond
 2025/02/20 05:50:01 Terraform apply | 
 2025/02/20 05:50:01 Terraform apply |   with module.code_engine_instance.ibm_code_engine_config_map.app_config_map["0"],
 2025/02/20 05:50:01 Terraform apply |   on ../modules/code-engine-project/main.tf line 87, in resource "ibm_code_engine_config_map" "app_config_map":
 2025/02/20 05:50:01 Terraform apply |   87: resource "ibm_code_engine_config_map" "app_config_map" {
 2025/02/20 05:50:01 Terraform apply | 
 2025/02/20 05:50:01 Terraform apply | The plugin encountered an error, and failed to respond to the
 2025/02/20 05:50:01 Terraform apply | plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain
 2025/02/20 05:50:01 Terraform apply | more details.
 2025/02/20 05:50:01 Terraform apply | 
 2025/02/20 05:50:01 Terraform apply | Error: Plugin did not respond
 2025/02/20 05:50:01 Terraform apply | 
 2025/02/20 05:50:01 Terraform apply |   with module.code_engine_instance.ibm_code_engine_app.app["0"],
 2025/02/20 05:50:01 Terraform apply |   on ../modules/code-engine-project/main.tf line 209, in resource "ibm_code_engine_app" "app":
 2025/02/20 05:50:01 Terraform apply |  209: resource "ibm_code_engine_app" "app" {
 2025/02/20 05:50:01 Terraform apply | 
 2025/02/20 05:50:01 Terraform apply | The plugin encountered an error, and failed to respond to the
 2025/02/20 05:50:01 Terraform apply | plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain
 2025/02/20 05:50:01 Terraform apply | more details.
 2025/02/20 05:50:01 Terraform apply | 
 2025/02/20 05:50:01 Terraform apply | Stack trace from the terraform-provider-ibm_v1.75.2 plugin:
 2025/02/20 05:50:01 Terraform apply | 
 2025/02/20 05:50:01 Terraform apply | panic: interface conversion: interface {} is []map[string]interface {}, not []interface {}
 2025/02/20 05:50:01 Terraform apply | 
 2025/02/20 05:50:01 Terraform apply | goroutine 93 [running]:
 2025/02/20 05:50:01 Terraform apply | github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/codeengine.ResourceIbmCodeEngineAppAppPatchAsPatch(0xc003383180?, 0xc003ba5780)
 2025/02/20 05:50:01 Terraform apply | 	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/codeengine/resource_ibm_code_engine_app.go:1369 +0xd06
 2025/02/20 05:50:01 Terraform apply | github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/codeengine.resourceIbmCodeEngineAppUpdate({0x6277ac8, 0xc0005fce70}, 0xc003ba5780, {0x584b7c0, 0xc003c2c008})
 2025/02/20 05:50:01 Terraform apply | 	github.com/IBM-Cloud/terraform-provider-ibm/ibm/service/codeengine/resource_ibm_code_engine_app.go:1140 +0x1659
 2025/02/20 05:50:01 Terraform apply | github.com/IBM-Cloud/terraform-provider-ibm/ibm/provider.wrapResource.wrapFunction.func5({0x6277ac8, 0xc0005fce70}, 0xc003ba5780, {0x584b7c0, 0xc003c2c008})
 2025/02/20 05:50:01 Terraform apply | 	github.com/IBM-Cloud/terraform-provider-ibm/ibm/provider/provider.go:1792 +0x143
 2025/02/20 05:50:01 Terraform apply | github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc00322a000, {0x6277a20, 0xc003c7d800}, 0xc003ba5780, {0x584b7c0, 0xc003c2c008})
 2025/02/20 05:50:01 Terraform apply | 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/resource.go:840 +0x119
 2025/02/20 05:50:01 Terraform apply | github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00322a000, {0x6277a20, 0xc003c7d800}, 0xc003ca9930, 0xc003ba5600, {0x584b7c0, 0xc003c2c008})
 2025/02/20 05:50:01 Terraform apply | 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/resource.go:947 +0x81a
 2025/02/20 05:50:01 Terraform apply | github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000844708, {0x6277a20?, 0xc003c7d740?}, 0xc003e674a0)
 2025/02/20 05:50:01 Terraform apply | 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.35.0/helper/schema/grpc_provider.go:1155 +0xd5c
 2025/02/20 05:50:01 Terraform apply | github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0007b8140, {0x6277a20?, 0xc003c7cf60?}, 0xc00004abd0)
 2025/02/20 05:50:01 Terraform apply | 	github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov5/tf5server/server.go:865 +0x3bc
 2025/02/20 05:50:01 Terraform apply | github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x5740980, 0xc0007b8140}, {0x6277a20, 0xc003c7cf60}, 0xc003ba4000, 0x0)
 2025/02/20 05:50:01 Terraform apply | 	github.com/hashicorp/terraform-plugin-go@v0.25.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:611 +0x1a6
 2025/02/20 05:50:01 Terraform apply | google.golang.org/grpc.(*Server).processUnaryRPC(0xc00004ca00, {0x6277a20, 0xc003c7ced0}, {0x6284f20, 0xc003336000}, 0xc003e87320, 0xc00327d260, 0x8a68378, 0x0)
 2025/02/20 05:50:01 Terraform apply | 	google.golang.org/grpc@v1.67.1/server.go:1394 +0xe2b
 2025/02/20 05:50:01 Terraform apply | google.golang.org/grpc.(*Server).handleStream(0xc00004ca00, {0x6284f20, 0xc003336000}, 0xc003e87320)
 2025/02/20 05:50:01 Terraform apply | 	google.golang.org/grpc@v1.67.1/server.go:1805 +0xe8b
 2025/02/20 05:50:01 Terraform apply | google.golang.org/grpc.(*Server).serveStreams.func2.1()
 2025/02/20 05:50:01 Terraform apply | 	google.golang.org/grpc@v1.67.1/server.go:1029 +0x7f
 2025/02/20 05:50:01 Terraform apply | created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 12
 2025/02/20 05:50:01 Terraform apply | 	google.golang.org/grpc@v1.67.1/server.go:1040 +0x125
 2025/02/20 05:50:01 Terraform apply | 
 2025/02/20 05:50:01 Terraform apply | Error: The terraform-provider-ibm_v1.75.2 plugin crashed!
 2025/02/20 05:50:01 Terraform apply | 
 2025/02/20 05:50:01 Terraform apply | This is always indicative of a bug within the plugin. It would be immensely
 2025/02/20 05:50:01 Terraform apply | helpful if you could report the crash with the plugin's maintainers so that it
 2025/02/20 05:50:01 Terraform apply | can be fixed. The output above should help diagnose the issue.
 2025/02/20 05:50:01 Terraform apply | 
 2025/02/20 05:50:01 [1m[31mTerraform APPLY error: Terraform APPLY errorexit status 1[39m[0m
 2025/02/20 05:50:01 [1m[31mCould not execute job: Error : Terraform APPLY errorexit status 1[39m[0m`

--- a/ibm/service/codeengine/resource_ibm_code_engine_app.go
+++ b/ibm/service/codeengine/resource_ibm_code_engine_app.go
@@ -1414,7 +1414,7 @@ func ResourceIbmCodeEngineAppAppPatchAsPatch(patchVals *codeenginev2.AppPatch, d
 	if _, exists := d.GetOk(path); d.HasChange(path) && !exists {
 		patch["run_env_variables"] = nil
 	} else if exists && patch["run_env_variables"] != nil {
-		ResourceIbmCodeEngineAppEnvVarPrototypeAsPatch(patch["run_env_variables"].([]interface{})[0].(map[string]interface{}), d)
+		ResourceIbmCodeEngineAppEnvVarPrototypeAsPatch(patch["run_env_variables"].(map[string]interface{}), d)
 	}
 	path = "run_service_account"
 	if _, exists := d.GetOk(path); d.HasChange(path) && !exists {
@@ -1424,7 +1424,7 @@ func ResourceIbmCodeEngineAppAppPatchAsPatch(patchVals *codeenginev2.AppPatch, d
 	if _, exists := d.GetOk(path); d.HasChange(path) && !exists {
 		patch["run_volume_mounts"] = nil
 	} else if exists && patch["run_volume_mounts"] != nil {
-		ResourceIbmCodeEngineAppVolumeMountPrototypeAsPatch(patch["run_volume_mounts"].([]interface{})[0].(map[string]interface{}), d)
+		ResourceIbmCodeEngineAppVolumeMountPrototypeAsPatch(patch["run_volume_mounts"].(map[string]interface{}), d)
 	}
 	path = "scale_concurrency"
 	if _, exists := d.GetOk(path); d.HasChange(path) && !exists {

--- a/ibm/service/codeengine/resource_ibm_code_engine_function.go
+++ b/ibm/service/codeengine/resource_ibm_code_engine_function.go
@@ -915,7 +915,7 @@ func ResourceIbmCodeEngineFunctionFunctionPatchAsPatch(patchVals *codeenginev2.F
 	if _, exists := d.GetOk(path); d.HasChange(path) && !exists {
 		patch["run_env_variables"] = nil
 	} else if exists && patch["run_env_variables"] != nil {
-		ResourceIbmCodeEngineFunctionEnvVarPrototypeAsPatch(patch["run_env_variables"].([]interface{})[0].(map[string]interface{}), d)
+		ResourceIbmCodeEngineFunctionEnvVarPrototypeAsPatch(patch["run_volume_mounts"].(map[string]interface{}), d)
 	}
 	path = "runtime"
 	if _, exists := d.GetOk(path); d.HasChange(path) && !exists {

--- a/ibm/service/codeengine/resource_ibm_code_engine_job.go
+++ b/ibm/service/codeengine/resource_ibm_code_engine_job.go
@@ -958,7 +958,7 @@ func ResourceIbmCodeEngineJobJobPatchAsPatch(patchVals *codeenginev2.JobPatch, d
 	if _, exists := d.GetOk(path); d.HasChange(path) && !exists {
 		patch["run_env_variables"] = nil
 	} else if exists && patch["run_env_variables"] != nil {
-		ResourceIbmCodeEngineJobEnvVarPrototypeAsPatch(patch["run_env_variables"].([]interface{})[0].(map[string]interface{}), d)
+		ResourceIbmCodeEngineJobEnvVarPrototypeAsPatch(patch["run_env_variables"].(map[string]interface{}), d)
 	}
 	path = "run_mode"
 	if _, exists := d.GetOk(path); d.HasChange(path) && !exists {
@@ -972,7 +972,7 @@ func ResourceIbmCodeEngineJobJobPatchAsPatch(patchVals *codeenginev2.JobPatch, d
 	if _, exists := d.GetOk(path); d.HasChange(path) && !exists {
 		patch["run_volume_mounts"] = nil
 	} else if exists && patch["run_volume_mounts"] != nil {
-		ResourceIbmCodeEngineJobVolumeMountPrototypeAsPatch(patch["run_volume_mounts"].([]interface{})[0].(map[string]interface{}), d)
+		ResourceIbmCodeEngineJobVolumeMountPrototypeAsPatch(patch["run_volume_mounts"].(map[string]interface{}), d)
 	}
 	path = "scale_array_spec"
 	if _, exists := d.GetOk(path); d.HasChange(path) && !exists {

